configure.ac

```
@@ -443,11 +443,33 @@ if test "x$with_dl" != "xno"; then
   AC_CHECK_LIB(dl, dlopen)
 fi
 
+AC_DEFUN([LIBATOMIC_CHECKER],[
+  AC_MSG_CHECKING(whether linking $1 -latomic works)
+  AC_LINK_IFELSE([
+    AC_LANG_SOURCE([[
+      #include <atomic>
+      struct LockRequired { char data[256]; };
+      int
+      main() {
```

Similarly at least the parameter types are needed for main:
```suggestion
      main(int, char **) {
```

---

Done, however, both variants [conform](https://en.cppreference.com/w/cpp/language/main_function) to the C++ standard.